tests/integration/test_flannel_integration.py

```diff
@@ -0,0 +1,138 @@
+    flannel_config = await flannel.get_config()
+    cidr_network = ip_network(flannel_config.get("cidr", {}).get("value"))
+
+    for unit in flannel.units:
+        assert unit.workload_status == "active"
+        assert _get_flannel_subnet_ip(unit) in cidr_network
+
+    # create test pod
+    resp = await _create_test_pod(ops_test.model)
+    assert ip_address(resp.status.pod_ip) in cidr_network, \
+        "the new pod does not get the ip address in the cidr network"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test, flannel_resource_name, flannel_resource):
+    """Build and deploy Flannel in bundle."""
+    flannel_charm = await ops_test.build_charm(".")
+
+    # Work around libjuju not handling local file resources by manually
+    # pre-deploying the charm w/ resource via the CLI. See
+    # https://github.com/juju/python-libjuju/issues/223
+    rc, stdout, stderr = await ops_test._run(
+        "juju",
+        "deploy",
+        "-m",
+        ops_test.model_full_name,
+        flannel_charm,
+        "--resource",
+        f"{flannel_resource_name}={flannel_resource}",  # noqa: E999
+    )
+    assert rc == 0, f"Failed to deploy with resource: {stderr or stdout}"
+
+    bundle = ops_test.render_bundle(
+        "tests/data/bundle.yaml",
+        master_charm=flannel_charm,
+        series="focal",
+        # flannel_resource_name=flannel_resource_name,  # This doesn't work currently
+        # flannel_resource=flannel_resource,  # This doesn't work currently
+    )
+    await ops_test.model.deploy(bundle)
+
+    # This configuration is needed due testing on top of LXD containers.
+    # https://bugs.launchpad.net/charm-kubernetes-worker/+bug/1903566
+    await ops_test.model.applications["kubernetes-worker"].set_config({
+        "kubelet-extra-config": "{protectKernelDefaults: false}"
+    })
+
+    await ops_test.model.wait_for_idle(wait_for_active=True, timeout=60 * 60,
+                                       idle_period=60)
+
+
+async def test_status_messages(ops_test):
+    """Validate that the status messages are correct."""
+    await validate_flannel_cidr_network(ops_test)
+
+
+async def test_change_cidr_network(ops_test):
+    """Test configuration change."""
+    flannel = ops_test.model.applications["flannel"]
+    await flannel.set_config({"cidr": "10.2.0.0/16"})
+    await ops_test.model.wait_for_idle(wait_for_active=True, idle_period=60)
+
+    # note (rgildein): There is need to restart kubernetes-worker machine.
+    #                  https://bugs.launchpad.net/charm-flannel/+bug/1932551
+    k8s_worker = ops_test.model.applications["kubernetes-worker"].units[0]
+    rc, stdout, stderr = await ops_test._run(
+        "ssh",
+        "-i", os.path.expanduser("~/.local/share/juju/ssh/juju_id_rsa"),
+        "-o", "StrictHostKeyChecking=no",
+        "-q",
+        "{}@{}".format("ubuntu", k8s_worker.private_address),
+        "sudo reboot now"
```

Comment:
Does `juju ssh` not work?  Also, I repeatedly find myself thinking that there is a `juju reboot` command and then remembering that it there isn't. :disappointed:

Reply:
Exactly the `juju reboot` display does not exist. The `juju ssh` should work :thinking:, I tried to use `libjuju` and `unit.ssh`, which does not work due to change IP.

Reply:
Even `juju ssh` has a problem (sometimes):
`AssertionError: Failed to restart kubernetes-worker with resource: ERROR unit "kubernetes-worker/0" not found`